pwndbg/glibc.py

```diff
@@ -68,11 +68,38 @@ def _get_version() -> Optional[Tuple[int, ...]]:
 @pwndbg.lib.memoize.reset_on_start
 @pwndbg.lib.memoize.reset_on_objfile
 def get_libc_filename_from_info_sharedlibrary() -> Optional[str]:
+    """
+    Get the filename of the libc by parsing the output of `info sharedlibrary`.
+    """
+    # Try to parse the output of `info sharedlibrary`:
+    # pwndbg> |info sharedlibrary| grep libc
+    # 0x00007f9ade418700  0x00007f9ade58f47d  Yes         ./libc.so.6
+    # Or:
+    # pwndbg> |info sharedlibrary| grep libc
+    # 0x00007f9ade418700  0x00007f9ade58f47d  Yes (*)     ./libc.so.6
+    possible_libc_path = []
     for line in pwndbg.gdblib.info.sharedlibrary().splitlines()[1:]:
-        filename = line.split(maxsplit=3)[-1].lstrip("(*)").lstrip()
-        # Is it possible that the libc is not called `libc.so.6`?
-        if os.path.basename(filename) == "libc.so.6":
-            return filename
+        if line.startswith("("):
+            # footer line:
+            # (*): Shared library is missing debugging information.
```

Comment:
https://github.com/bminor/binutils-gdb/blob/cb8d98ec9251bb67f9b394c63f8c350a5be1a272/gdb/solib.c#L1153-L1161

Seems this is the only message that starts with `(*)`; a potential problem here is that the call in GDB code is`_("...")` and that `_` is probably internalization/translation stuff and the translated string in theory may not start with `(*)` (but I doubt they would change it).

Just saying. I guess we may not be able to do better than that.